router/worker.go

```diff
@@ -774,27 +778,41 @@ func (w *worker) proxyRequest(ctx context.Context, destinationJob types.Destinat
 		Adapter: transformer.NewTransformerProxyAdapter(w.rt.transformerFeaturesService.TransformerProxyVersion(), w.rt.logger),
 	}
 	rtlTime := time.Now()
+	destination := destinationJob.Destination
+	destinationInfo := &oauthv2.DestinationInfo{
+		Config:           destination.Config,
+		DefinitionConfig: destination.DestinationDefinition.Config,
+		WorkspaceID:      destination.WorkspaceID,
+		DefinitionName:   destination.DestinationDefinition.Name,
+		ID:               destination.ID,
+	}
+	ctx = context.WithValue(ctx, common.DestKey, destinationInfo)
+	oauthV2Enabled := w.rt.reloadableConfig.oauthV2Enabled.Load()
 	proxyRequestResponse := w.rt.transformer.ProxyRequest(ctx, proxyReqparams)
 	w.routerProxyStat.SendTiming(time.Since(rtlTime))
 	w.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request ended`, w.rt.destType, jobID)
 	authType := oauth.GetAuthType(destinationJob.Destination.DestinationDefinition.Config)
-	if proxyRequestResponse.ProxyRequestStatusCode != http.StatusOK && routerutils.IsNotEmptyString(string(authType)) && authType == oauth.OAuth {
-		w.logger.Debugf(`Sending for OAuth destination`)
-		// Token from header of the request
-		respStatusCode, respBodyTemp, contentType := w.rt.handleOAuthDestResponse(&HandleDestOAuthRespParams{
-			ctx:            ctx,
-			destinationJob: destinationJob,
-			workerID:       w.id,
-			trRespStCd:     proxyRequestResponse.ProxyRequestStatusCode,
-			trRespBody:     proxyRequestResponse.ProxyRequestResponseBody,
-			secret:         m[0].Secret,
-			contentType:    proxyRequestResponse.RespContentType,
-		}, proxyRequestResponse.OAuthErrorCategory)
-
-		proxyRequestResponse.RespStatusCodes, proxyRequestResponse.RespBodys = w.prepareResponsesForJobs(&destinationJob, respStatusCode, respBodyTemp)
-		proxyRequestResponse.RespContentType = contentType
+	if authType == oauth.OAuth {
+		if proxyRequestResponse.ProxyRequestStatusCode != http.StatusOK && !oauthV2Enabled {
```

Comment:
We can probably return early if it is not oauth type
```suggestion
		if authType != oauth.OAuth {
		return proxyRequestResponse
	}

	if proxyRequestResponse.ProxyRequestStatusCode != http.StatusOK && !oauthV2Enabled {d {
```

Reply:
Can you once check if the changes look good ?